Source: Gentoo, Diego Petteno
Upstream: Unknown
Reason: Fix zlib test suite failure, segfaults (often when libxml2 calls zlib functions) on 64 bit

Comments:
Try two at guessing what zlib wanted to do, assume a mistake between .4 and .5
where a condition was reversed, this one makes more sense in light of 1.2.4,
the definitions _and_ the comments.

Index: zlib-1.2.5/zlib.h
===================================================================
--- zlib-1.2.5.orig/zlib.h
+++ zlib-1.2.5/zlib.h
@@ -1578,7 +1578,7 @@ ZEXTERN int ZEXPORT inflateBackInit_ OF(
 #  define gzoffset gzoffset64
 #  define adler32_combine adler32_combine64
 #  define crc32_combine crc32_combine64
-#  ifdef _LARGEFILE64_SOURCE
+#  ifndef _LARGEFILE64_SOURCE
      ZEXTERN gzFile ZEXPORT gzopen64 OF((const char *, const char *));
      ZEXTERN z_off_t ZEXPORT gzseek64 OF((gzFile, z_off_t, int));
      ZEXTERN z_off_t ZEXPORT gztell64 OF((gzFile));
